File: agent-driven-architecture-main/examples/kotlin/block/console/src/main/kotlin/adr/blocks/console/Slice.kt
// ── blocks/console/slice — presentation TRUTH, folded like any other ───────
// 4.6's test, verbatim: if losing the field on a re-fold would change what the
// system believes or what the artifact contains, it is truth — fold it. Which panel
// the agent decided to open is a decision someone made and a reader may need to
// audit ("why did the escalation button disappear?"). It folds.

package adr.blocks.console

import adr.spine.pure.PanelId
import adr.spine.pure.TicketId

data class ConsoleSlice(
    val focused: TicketId? = null,
    val panels: Map<PanelId, Boolean> = emptyMap(),
) {
    // No companion: a companion member has no instance, the same defect as a top-level
    // function. The EMPTY slice is what the primary constructor builds when told
    // nothing — `ConsoleSlice()` — so the shape carries its own starting value.
    fun withFocus(ticket: TicketId): ConsoleSlice = copy(focused = ticket)

    fun withPanel(panel: PanelId, visible: Boolean): ConsoleSlice =
        copy(panels = panels + (panel to visible))
}

File: agent-driven-architecture-main/examples/typescript/src/blocks/console/slice.ts
// ── blocks/console/slice — presentation TRUTH, folded like any other ───────
// Which ticket is focused and which panels are shown are decisions someone
// made; losing them on a re-fold would change what the system believes about
// the session, so by 4.6's own test they belong in State.

import type { PanelId, TicketId } from "@adr/spine/pure/ids";

export interface ConsoleSlice {
  readonly focused: TicketId | null;
  readonly panels: ReadonlyMap<PanelId, boolean>;
}

export const emptyConsoleSlice: ConsoleSlice = { focused: null, panels: new Map() };

export function consoleSliceOf(panels: readonly PanelId[]): ConsoleSlice {
  return { focused: null, panels: new Map(panels.map((p) => [p, true])) };
}

export function knowsPanel(slice: ConsoleSlice, panel: PanelId): boolean {
  return slice.panels.has(panel);
}

export function withFocus(slice: ConsoleSlice, ticket: TicketId): ConsoleSlice {
  return { ...slice, focused: ticket };
}

export function withPanel(slice: ConsoleSlice, panel: PanelId, visible: boolean): ConsoleSlice {
  const panels = new Map(slice.panels);
  panels.set(panel, visible);
  return { ...slice, panels };
}
